substrafl/remote/register/manage_dependencies.py

```diff
@@ -0,0 +1,206 @@
+    """Generates wheels for the private modules from lib_modules list and returns the list of names for each wheel.
+
+     It first creates the wheel for each library. Each of the libraries must be already installed in the correct
+     version locally. Use command: ``pip install -e library-name`` in the directory of each library.
+     Then it copies the wheels to the dest_dir.
+
+    This allows one user to use custom version of the passed modules.
+
+    Args:
+        lib_modules (list): list of modules to be installed.
+        dest_dir (str): relative directory where the wheels are saved
+
+    Returns:
+        List[str]: wheel names for the given modules
+    """
+    wheel_names = []
+    dest_dir.mkdir(exist_ok=True, parents=True)
+    for lib_module in lib_modules:
+        if not (Path(lib_module.__file__).parents[1] / "setup.py").exists():
+            msg = ", ".join([lib.__name__ for lib in lib_modules])
+            raise NotImplementedError(
+                f"You must install {msg} in editable mode.\n" "eg `pip install -e substra` in the substra directory"
+            )
+        lib_name = lib_module.__name__
+        lib_path = Path(lib_module.__file__).parents[1]
+        wheel_name = f"{lib_name}-{lib_module.__version__}-py3-none-any.whl"
+
+        wheel_path = LOCAL_WHEELS_FOLDER / wheel_name
+        # Recreate the wheel only if it does not exist
+        if wheel_path.exists():
+            logger.warning(
+                f"Existing wheel {wheel_path} will be used to build {lib_name}. "
+                "It may lead to errors if you are using an unreleased version of this lib: "
+                "if it's the case, you can delete the wheel and it will be re-generated."
+            )
+        else:
+            # if the right version of substra or substratools is not found, it will search if they are already
+            # installed in 'dist' and take them from there.
+            # sys.executable takes the Python interpreter run by the code and not the default one on the computer
+            extra_args: list = []
+            if lib_name == "substrafl":
+                extra_args = [
+                    "--find-links",
+                    dest_dir.parent / "substra",
+                    "--find-links",
+                    dest_dir.parent / "substratools",
+                ]
+            subprocess.check_output(
+                [
+                    sys.executable,
+                    "-m",
+                    "pip",
+                    "wheel",
+                    ".",
+                    "-w",
+                    LOCAL_WHEELS_FOLDER,
+                    "--no-deps",
+                ]
+                + extra_args,
+                cwd=str(lib_path),
+            )
+
+        shutil.copy(wheel_path, dest_dir / wheel_name)
+        wheel_names.append(wheel_name)
+
+    return wheel_names
+
+
+def substra_libraries_pypi_dependencies(lib_modules: List) -> List[str]:
```

Comment:
The function name is a bit confusing as the function is not a all substra dependent. It just parse the name and version of a given list of installed dep... 
I would put it as a private utils function and call it "_extract_dependencies_informations" or smthg like this